Source/WebCore/dom/ContainerNodeAlgorithms.cpp

```diff
@@ -110,9 +110,11 @@ static RemovedSubtreeObservability notifyNodeRemovedFromDocument(ContainerNode&
     RemovedSubtreeObservability observability = RemovedSubtreeObservability::NotObservable;
     for (RefPtr currentNode = &node; currentNode; currentNode = NodeTraversal::next(*currentNode)) {
         currentNode->removedFromAncestor(Node::RemovalType { /* disconnectedFromDocument */ true, treeScopeChange == TreeScopeChange::Changed }, oldParentOfRemovedTree);
+        if (isCalledFromRemoveChildren == IsCalledFromRemoveChildren::Yes)
+            node.document().childrenToBeDeleted().approximateMemoryStored += currentNode->approximateMemoryCost();
```

Comment:
Aren't other parts of the codebase which depend on the (currently incomplete) implementation of `approximateMemoryCost` (GC?) ?

We should indeed fix that, but it could certainly be an independent/followup patch.